Microsoft.Azure.Cosmos/src/Query/v3Query/CosmosQueryClientCore.cs

```
@@ -323,9 +324,19 @@ public override void ClearSessionTokenCache(string collectionFullName)
                     }
 
                     long responseLengthBytes = memoryStream.Length;
-                    CosmosArray documents = CosmosQueryClientCore.ParseElementsFromRestStream(
+                    CosmosQueryClientCore.ParseRestStream(
                         memoryStream,
-                        resourceType);
+                        resourceType,
+                        out CosmosArray documents,
+                        out CosmosObject distributionPlan);
+
+                    DistributionPlanPayload distributionPlanPayload = null;
+
+                    if (distributionPlan.TryGetValue("backendDistributionPlan", out CosmosElement backendDistributionPlan) &&
```

We can rewrite the code to reflect the contract better and fail early/fast when it's not honored.
For e.g. if both or none of the fields must be present, I would assert according to that.

---

Is this addressed? I still see the original, somewhat ambiguous implementation.